gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/SSOSessionHandler.java

```diff
@@ -194,12 +198,15 @@ private void checkClient(RoutingContext context, io.gravitee.am.model.User user,
                         return Completable.complete();
                     }
 
+                    Set<String> userIdentities = ofNullable(user.getIdentities()).map(identities -> identities.stream().map(UserIdentity::getProviderId).collect(Collectors.toSet())).orElse(new HashSet<>());
+                    userIdentities.add(user.getSource());
```

Comment:
question: instead of stream in stream maybe more flatten form is more readable? WDYT?
`ofNullable(user.getIdentities()).orElseGet(List::of)
                            .stream()
                            .map(UserIdentity::getProviderId)
                            .collect(Collectors.toSet());`